tests/e2e/kafka.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireDeleteKafkaTopic deletes kafkaTopic resource.
+func requireDeleteKafkaTopic(kubectlOptions k8s.KubectlOptions, topicName string) {
+	It("Deleting KafkaTopic CR", func() {
+		deleteK8sResource(kubectlOptions, defaultDeletionTimeout, kafkaTopicKind, "", topicName)
+	})
+}
+
+// requireDeployingKafkaTopic deploys a kafkaTopic resource from a template
+func requireDeployingKafkaTopic(kubectlOptions k8s.KubectlOptions, topicName string) {
+	It("Deploying KafkaTopic CR", func() {
+		applyK8sResourceFromTemplate(kubectlOptions,
+			kafkaTopicTemplate,
+			map[string]interface{}{
+				"Name":      topicName,
+				"TopicName": topicName,
+				"Namespace": kubectlOptions.Namespace,
+			},
+		)
```

Comment:
**Required**: We **MUST** check potential errors here.

```suggestion
		err := applyK8sResourceFromTemplate(kubectlOptions,
			kafkaTopicTemplate,
			map[string]interface{}{
				"Name":      topicName,
				"TopicName": topicName,
				"Namespace": kubectlOptions.Namespace,
			},
		)
		
		Expect(err).NotTo(HaveOccurred())
```

Reply:
fixed: [3efc5f9](https://github.com/banzaicloud/koperator/pull/987/commits/3efc5f92e73093a477784196d6bcf64ffcc0a417)